modules/renter.go

```diff
@@ -293,7 +293,7 @@ type Renter interface {
 
 	// Download performs a download according to the parameters passed, including
 	// downloads of `offset` and `length` type.
-	Download(params RenterDownloadParameters) error
+	Download(params RenterDownloadParameters) chan error
```

Comment:
I am hesitant to approve of a change like this, especially in a pull request that's dedicated mostly to testing.

Was there any particular reason you felt it was necessary?

Reply:
Yes, without this change there is a race condition between the Download api call and the downloadqueue api call. Previously if you called Download, there was no guarantee that the download actually made it into the downloadqueue before the next call to downloadqueue. That's because we only waited for a fixed time of 100ms instead of returning a channel.